src/views/product-landing/helpers/validate-against-schema.ts

```diff
@@ -0,0 +1,35 @@
+import { Schema, ValidationErrorItem } from 'joi'
+
+export function validateAgainstSchema<T>(
+  data: unknown,
+  JoiSchema: Schema,
+  sourceFile: string
+): asserts data is T {
```

Comment:
Update relative to assembled PRs: I realized I could do `asserts data is T` as the return type here. Compared to returning a boolean (`data is T)`, this avoids having to check with an `if` statement after the type guard. [This Stackoverflow answer](https://stackoverflow.com/a/49696654) helped me out.